src/theory/datatypes/sygus_extension.cpp

```diff
@@ -1101,16 +1102,20 @@ Node SygusExtension::registerSearchValue(Node a,
         if (bv != bvr)
         {
           // add to the sampler database object
-          std::map<TypeNode, quantifiers::SygusSampler>::iterator its =
-              d_sampler[a].find(tn);
-          if (its == d_sampler[a].end())
+          std::map<TypeNode, std::unique_ptr<quantifiers::SygusSampler>>& smap =
+              d_sampler[a];
+          std::map<TypeNode,
+                   std::unique_ptr<quantifiers::SygusSampler>>::iterator its =
+              smap.find(tn);
+          if (its == smap.end())
           {
-            d_sampler[a][tn].initializeSygus(
+            smap[tn].reset(new quantifiers::SygusSampler(d_env.getRewriter()));
```

Comment:
It is not necessary to change this to a `std::unique_ptr`, I think. We can potentially do:

```
smap[tn].emplace(tn, d_env.getRewriter());
```

Reply:
Actually, i believe this does not work.  The second argument to emplace is the value of the range, but I don't think it can determine that this is the argument to the constructor of the range. Think if the sampler had multiple arguments in its constructor.